plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/NativeImage.kt

```
@@ -429,48 +429,50 @@ open class NativeImage @Inject constructor(
+                        + "-H:NativeLinkerOption=-Wl,--no-whole-archive"
+                        + librariesOptions()
+                        + sgxLibrariesOptions()
+                        + linkerScriptOption()
+                        + reflectConfigurationOption()
+                        + includeResourcesOption(appResourcesConfig.get().asFile)
+                        + serializationConfigurationOption()
+                        + getLanguages()
             )
-            + (if (enclaveMode != EnclaveMode.RELEASE) debugOptions else emptyList())
-            + defaultOptions()
-            + compilerOptions
-            + placeholderLibPathOption()
-            + includePathsOptions()
-            + libraryPathOptions()
-            + "-H:NativeLinkerOption=-Wl,--whole-archive"
-            + librariesWholeArchiveOptions()
-            + "-H:NativeLinkerOption=-Wl,--no-whole-archive"
-            + librariesOptions()
-            + sgxLibrariesOptions()
-            + linkerScriptOption()
-            + reflectConfigurationOption()
-            + includeResourcesOption(appResourcesConfig.get().asFile)
-            + serializationConfigurationOption()
-            + getLanguages()
-        )
-        if (errorOut?.any { "Image generator watchdog is aborting image generation" in it } == true) {
-            // If there is too much memory pressure in the container, native image can just get slower and slower until
-            // a watchdog timer fires. Give the user some advice on how to fix it.
-            linuxExec.throwOutOfMemoryException()
-        }
-        else if (errorOut?.any { "Default native-compiler executable 'gcc' not found via environment variable PATH" in it } == true) {
-            // This error will only occur on Linux because on other platforms native-image is run in a docker container that
-            // already has gcc installed.
-            throw GradleException(
+        } catch (e: GradleException) {
```

Not sure why before we were relying on the return of `LinuxExec.exec` to handle the errors. I feel it is more correct to throw the exception anyway (avoiding swallow). And now that I actually need the output, I think it is nicer to catch and rethrow in case of exceptions.

---

I agree